src/Control/Effect/NonDet.hs

```diff
@@ -12,81 +12,121 @@ module Control.Effect.NonDet
 , run
 ) where
 
-import Control.Applicative (Alternative(..))
+import Control.Applicative (Alternative(..), liftA2)
 import Control.Effect.Carrier
-import Control.Monad (MonadPlus(..))
+import Control.Monad (MonadPlus(..), join)
 import qualified Control.Monad.Fail as Fail
 import Control.Monad.Fix
 import Control.Monad.IO.Class
 import Control.Monad.Trans.Class
+import Data.Maybe (fromJust)
 import GHC.Generics (Generic1)
 
 data NonDet m k
   = Empty
   | Choose (Bool -> m k)
-  deriving stock (Functor, Generic1)
-  deriving anyclass (HFunctor, Effect)
+  deriving (Functor, Generic1)
+
+instance HFunctor NonDet
+instance Effect   NonDet
```

Comment:
I honestly don’t think we should use `DeriveAnyClass` for these, as it’s a false economy.

Reply:
Up to you. I personally like it, but given that it’s a two-line difference I’m not worked up about it.